solutions/instances/variables.tf

```diff
@@ -43,14 +43,147 @@ variable "prefix" {
+  }
+}
+
+variable "enable_cloud_logs_data" {
+  type        = bool
+  description = "Whether to enable cloud logs data to the Object Storage bucket. If set to true, `cloud_logs_provision` must also be set to true."
+  default     = true
+}
+
+variable "enable_en_cloud_logs_integration" {
+  type        = bool
+  description = "Whether to enable event notification integration for IBM Cloud Logs instance. When set to true, a value is required for `existing_en_instance_crn`."
+  default     = false
+}
+
+variable "existing_en_instance_crn" {
+  type        = string
+  description = "The CRN of the existing event notification instance. If a value is provided here, `enable_en_cloud_logs_integration` must be set to true in order to enable the integration."
+  default     = null
+}
+
+variable "en_integration_name" {
+  type        = string
+  description = "The name of the event notification integration that gets created. If a prefix input variable is passed, it is prefixed to the value in the `<prefix>-value` format."
+  default     = "cloud-logs-en-integration"
+}
+
+variable "skip_en_auth_policy" {
+  type        = bool
+  description = "To skip creating auth policy that allows Cloud Logs 'Event Source Manager' role access in the existing event notification instance."
+  default     = false
+}
+
+variable "cloud_logs_retention_period" {
+  type        = number
+  description = "The number of days IBM Cloud Logs will retain the logs data in priority insights. Possible Values: 7, 14, 30, 60, 90"
+  default     = 7
+
+  validation {
+    condition     = contains([7, 14, 30, 60, 90], var.cloud_logs_retention_period)
+    error_message = "The retention period must be one of the following values: 7, 14, 30, 60, or 90 days."
+  }
+}
+
+variable "cloud_log_data_bucket_name" {
+  type        = string
+  default     = "cloud-logs-data-bucket"
+  description = "The name of the Cloud Object Storage bucket to create to store cloud log data. Cloud Object Storage bucket names are globally unique. If the `add_bucket_name_suffix` variable is set to `true`, 4 random characters are added to this name to ensure that the name of the bucket is globally unique. If the prefix input variable is passed, the name of the bucket is prefixed to the value in the `<prefix>-value` format."
+}
+
+variable "existing_cloud_logs_data_bucket_crn" {
+  type        = string
+  nullable    = true
+  default     = null
+  description = "The crn of an existing bucket within the Cloud Object Storage instance to store IBM Cloud Logs data. If an existing Cloud Object Storage bucket is not specified, a bucket is created."
+}
+
+variable "existing_cloud_logs_data_bucket_endpoint" {
+  type        = string
+  nullable    = true
+  default     = null
+  description = "The name of an existing Cloud Object Storage bucket endpoint to use for storing the IBM Cloud Logs data. If an existing endpoint is not specified, the endpoint of the new Cloud Object Storage bucket is used."
```

Comment:
"the endpoint of the new Cloud Object Storage bucket is used" <-- I don't understand this. If user is passing an existing bucket, there would be no new bucket created here?

Reply:
Yes, if user is passing existing bucket details then `cloud_log_data_bucket` will not be created. 

Also, updated the description to "`The endpoint of an existing Cloud Object Storage bucket to use for storing the IBM Cloud Logs data. If an existing Cloud Object Storage bucket is not specified, a bucket is created.`"